File: LeetCode_Solutions/LeetCode_Solutions/NumberComplement_476.m
//
//  NumberComplement_476.m
//  LeetCode_Solutions
//
//  Created by Huni on 20/03/2017.
//  Copyright © 2017 KnightJoker. All rights reserved.
//



#import "NumberComplement_476.h"

@implementation NumberComplement_476

- (NSInteger)findComplement:(NSInteger)num {

    NSInteger temp = num, mask = 1;
    while(temp) {
            temp >>= 1;
            mask <<= 1;
        }
        return ((mask - 1) ^ num);
}


@end

File: LeetCode_Swift/LeetCode_Swift/NumberComplement_476.swift
//
//  NumberComplement_476.swift
//  LeetCode_Swift
//
//  Created by Huni on 25/08/2017.
//  Copyright © 2017 KnighhtJoker. All rights reserved.
//

import Foundation

class NumberComplement_476 {
    func findComplement(_ num: Int) -> Int {
        
        var temp = num,mask = 1
        while temp != 0 {
            temp >>= 1
            mask <<= 1
        }
        return (mask - 1) ^ num
    }
}
